### DISCUSSION TITLE
It's a SolidJS problem?

### DISCUSSION THREAD

**Discussion by sudonite** [solidjs/solid#2092] [Q&A]
Almost every package that I try to use is broken. I'm just importing one component. But every other is requested also, and its slow to load the page every time. Is this a solidjs problem or could the error be in the 3rd party packages?

### Package:  @ark-ui/solid:
**_It's finished, but after 15 sec loading._**

![2024-03-08T03:00:25,426506465+01:00](https://github.com/solidjs/solid/assets/71099189/5ff4e5f9-5f50-4086-8ffd-f23794f4d818)

### Package: @kobalte/core: 
**_Same as above the ark-ui._**

![2024-03-08T03:49:22,758193818+01:00](https://github.com/solidjs/solid/assets/71099189/4194a72f-2570-4083-87a4-8c9aa51c6fe2)

### Package: lucide-solid
**_Page never loaded, tab froze up. And this shows log warning's._**

![2024-03-08T03:03:38,301116936+01:00](https://github.com/solidjs/solid/assets/71099189/ba254491-e210-438d-8965-425c040b48bb)
![2024-03-08T03:03:59,103566967+01:00](https://github.com/solidjs/solid/assets/71099189/0fa23c1d-bd49-4619-adea-be7d2151efd8)

### Package: solid-icons
**_Its working without any problem._**

![2024-03-08T03:01:53,976878351+01:00](https://github.com/solidjs/solid/assets/71099189/25f73120-df7f-4d99-83a4-0a6eb60a0ad8)

The project created by official docs:
```
npx degit solidjs/templates/js my-app && cd my-app
npm install
npm install @ark-ui/solid
npm install @kobalte/core
npm install lucide-solid
npm install solid-icons
npm run dev
```

Just the App.jsx componenet is modified as shown in the pictures, every other file is untouched.

```
$ node -v
v21.7.0
```

```
$ npm -v
10.5.0
```

```
$ cat package.json
{
  "name": "vite-template-solid",
  "version": "0.0.0",
  "description": "",
  "scripts": {
    "start": "vite",
    "dev": "vite",
    "build": "vite build",
    "serve": "vite preview"
  },
  "license": "MIT",
  "devDependencies": {
    "solid-devtools": "^0.29.2",
    "vite": "^5.0.11",
    "vite-plugin-solid": "^2.8.2"
  },
  "dependencies": {
    "@ark-ui/solid": "^2.2.0",
    "@kobalte/core": "^0.12.3",
    "lucide-solid": "^0.350.0",
    "solid-icons": "^1.1.0",
    "solid-js": "^1.8.11"
  }
}
```

### Edit 1:
**@suid/material** is same as  **@ark-ui/solid** and **@kobalte/core**. I'm a backend, pentest person, trying to switch from React to Solid, without success. Maybe I'm doing something really wrong.

### Edit 2:
**@zag-js/solid** same. It's not surprising, because it's a **@ark-ui/solid** dependecy package.

**Comment by ryansolid** [comment]
In dev Vite has the potential to pull in all the dependencies. I haven't done much development with those libraries. Maybe someone with experience here can talk to this. I'm gathering this problem doesn't exist when prod built/served.

**Comment by ghost-2e63** [comment]
I tried and able to reproduce the same error with `@kobalte/core` , it is indeed fetching everything else, when importing just one component atleast in the dev build, while i can confirm that the prod build worked just fine. 

EDIT: Possibly related to this https://github.com/vitejs/vite/issues/8237 (Thanks @Brendonovich for linking this in discord.)

**Comment by sudonite** [comment]
Any solution or workaround? Or any step to skip ViteJS? Manualy crafted config files to bundler, etc. Solid-start metaframefork is okay to be solve the problem?

**Comment by sudonite** [comment]
https://github.com/solidjs/solid-start/issues/1048#issuecomment-1869677739
